api/types/accesslist/accesslist.go

```diff
@@ -538,3 +538,122 @@ func (a *AccessList) setInitialAuditDate(clock clockwork.Clock) {
 	a.Spec.Audit.NextAuditDate = clock.Now()
 	a.Spec.Audit.NextAuditDate = a.SelectNextReviewDate()
 }
+
+// NormalizeSemanticallyEqualFields normalizes semantically equal fields in the AccessList
+// to ensure that they are strictly equal when compared.
+func (a *AccessList) NormalizeSemanticallyEqualFields(b *AccessList) {
+	normalizeSlicesIfEqual(&a.Spec.MembershipRequires.Roles, &b.Spec.MembershipRequires.Roles)
+	normalizeSlicesIfEqual(&a.Spec.OwnershipRequires.Roles, &b.Spec.OwnershipRequires.Roles)
+	normalizeSlicesIfEqual(&a.Spec.Grants.Roles, &b.Spec.Grants.Roles)
+	normalizeSlicesIfEqual(&a.Spec.OwnerGrants.Roles, &b.Spec.OwnerGrants.Roles)
+	normalizeTraitsIfEqual(&a.Spec.Grants.Traits, &b.Spec.Grants.Traits)
+	normalizeTraitsIfEqual(&a.Spec.OwnerGrants.Traits, &b.Spec.OwnerGrants.Traits)
+	normalizeTraitsIfEqual(&a.Spec.MembershipRequires.Traits, &b.Spec.MembershipRequires.Traits)
+	normalizeTraitsIfEqual(&a.Spec.OwnershipRequires.Traits, &b.Spec.OwnershipRequires.Traits)
+}
+
+// IsEqual returns whether the Access List is equal to another Access List, ignoring
+// ephemeral fields such as `Status`, `Metadata.Revision`, and `IneligibleStatus` on Owners.
+func (a *AccessList) IsEqual(b *AccessList) bool {
```

Comment:
Q: Is this a valid way to go about adding this new logic? In theory, I think this makes more sense for validating an access list is == to another when calling `IsEqual`, but in terms of backwards-compat, `IsEqual` is called in quite a few other places currently.